framework/src/main/java/org/checkerframework/framework/util/JavaParserUtil.java

```diff
@@ -276,4 +308,39 @@ public void visit(BinaryExpr node, Void p) {
       }
     }
   }
+
+  /**
+   * Returns the {@link com.github.javaparser.ParserConfiguration.LanguageLevel} corresponding to
+   * the current source version.
+   *
+   * @param env processing environment used to get source version
+   * @return the current source version
+   */
+  public static ParserConfiguration.LanguageLevel getCurrentSourceVersion(
```

Comment:
Can the result of this method be cached?
It is far from our biggest performance problem, but the recomputation does seem needless.